cmake/kokkos_enable_options.cmake

```diff
@@ -68,11 +70,20 @@ FUNCTION(check_device_specific_options)
         ENDIF()
       ENDIF()
       IF(KOKKOS_ENABLE_${OPTION})
-        MESSAGE(FATAL_ERROR "Kokkos_ENABLE_${OPTION} is ON but ${SOME_DEVICE} backend is not enabled.")
+        IF (KOKKOS_SKIP_VALIDATION)
+          MESSAGE(STATUS "${SOME_DEVICE} is not active - turning off ${OPTION}")
+          SET(KOKKOS_ENABLE_${OPTION} OFF PARENT_SCOPE) #turn it off, don't error out
+        ELSE()
+          MESSAGE(FATAL_ERROR "Kokkos_ENABLE_${OPTION} is ON but ${SOME_DEVICE} backend is not enabled.")
+        ENDIF()
       ENDIF()
     ENDFOREACH()
   ENDIF()
 ENDFUNCTION()
 
 CHECK_DEVICE_SPECIFIC_OPTIONS(DEVICE CUDA OPTIONS CUDA_UVM CUDA_RELOCATABLE_DEVICE_CODE CUDA_LAMBDA CUDA_LDG_INTRINSIC)
 CHECK_DEVICE_SPECIFIC_OPTIONS(DEVICE HPX OPTIONS HPX_ASYNC_DISPATCH)
+
+IF (KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION)
+  SET(KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION ON)
+ENDIF()
```

Comment:
Add a comment that we the macro `KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION` was deprecated and that Kokkos expect the other name.